src/qibolab/instruments/rfsoc.py

```diff
@@ -1,54 +1,156 @@
+    """Convert `qibolab.sweeper.Sweeper` to `qibosoq.abstract.Sweeper`"""
+
+    parameters = []
+    starts = []
+    stops = []
+    indexes = []
+
+    if sweeper.parameter is Parameter.bias:
+        for qubit in sweeper.qubits:
+            parameters.append(rfsoc.Parameter.bias)
+            for idx, seq_qubit in enumerate(qubits):
+                if qubit == seq_qubit:
+                    indexes.append(idx)
+            starts.append(sweeper.values[0] + qubits[qubit].flux.bias)
+            stops.append(sweeper.values[-1] + qubits[qubit].flux.bias)
+
+        if max(np.abs(starts)) > 1 or max(np.abs(stops)) > 1:
+            raise ValueError("Sweeper amplitude is set to reach values higher than 1")
+    else:
+        for pulse in sweeper.pulses:
+            for idx, seq_pulse in enumerate(sequence):
+                if pulse == seq_pulse:
+                    indexes.append(idx)
+
+            if sweeper.parameter is Parameter.amplitude:
+                parameters.append(rfsoc.Parameter.amplitude)
+                starts.append(sweeper.values[0] * pulse.amplitude)
+                stops.append(sweeper.values[-1] * pulse.amplitude)
+            elif sweeper.parameter is Parameter.relative_phase:
+                parameters.append(rfsoc.Parameter.relative_phase)
+                starts.append(np.degrees(sweeper.values[0] + pulse.relative_phase))
+                stops.append(np.degrees(sweeper.values[-1] + pulse.relative_phase))
+            elif sweeper.parameter is Parameter.frequency:
+                # frequency are not convertet yet and LO is not accounted for
+                parameters.append(rfsoc.Parameter.frequency)
+                starts.append(sweeper.values[0] + pulse.frequency)
+                stops.append(sweeper.values[-1] + pulse.frequency)
+
+    return rfsoc.Sweeper(
+        parameter=parameters,
+        indexes=indexes,
+        starts=np.array(starts),
+        stops=np.array(stops),
+        expts=len(sweeper.values),
+    )
+
+
+class RFSoC(AbstractInstrument):
+    """Instrument object for controlling RFSoC FPGAs.
     The two way of executing pulses are with ``play`` (for arbitrary
     qibolab ``PulseSequence``) or with ``sweep`` that execute a
     ``PulseSequence`` object with one or more ``Sweeper``.
 
-    Args:
-        name (str): Name of the instrument instance.
     Attributes:
-        cfg (QickProgramConfig): Configuration dictionary required for pulse execution.
-        soc (QickSoc): ``Qick`` object needed to access system blocks.
+        cfg (rfsoc.Config): Configuration dictionary required for pulse execution.
     """
 
     def __init__(self, name: str, address: str):
+        """__init__
+        Args:
+            name (str): Name of the instrument instance.
+            address (str): IP and port of the server (ex. 192.168.0.10:6000)
+        """
+
         super().__init__(name, address=address)
         self.host, self.port = address.split(":")
```

Comment:
Since you have to split, maybe you can ask `host` and `port` separately. In this way the `port` could be simply an integer, that is a more appropriate type.